### PR TITLE
🗞️ Migrate the OApp standard into devTools

### PR DESCRIPTION
**Pull request by TRileySchwarz**

### CODE REVIEW

File: packages/oapp-evm/.gitignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
+out
```
Comment (janjakubnanista) [inline packages/oapp-evm/.gitignore:1]: Drop this if you still have some committing energy

File: packages/oapp-evm/foundry.toml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+[profile.default]
+solc = '0.8.22'
+verbosity = 3
+src = "contracts"
+test = "test"
+out = "artifacts"
+cache_path = "cache"
+optimizer = true
+optimizer_runs = 20_000
```
Comment (janjakubnanista) [inline packages/oapp-evm/foundry.toml:9]: You can speed up the build by removing this or setting it to like 200 - we are not deploying these directly so having the optimizer here does not matter at all

File: packages/oapp-evm/turbo.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  "extends": ["//"],
+  "pipeline": {
+    "build": {
+      "outputs": [
+        "dist/**",
+        "artifacts*/**",
+        "hh-cache*/**",
+        "cache*/**",
+        "src/typechain-types/**",
+        "out/**",
```
Comment (janjakubnanista) [inline packages/oapp-evm/turbo.json:11]: These can be dropped along with the `hh-cache`

File: packages/oapp-evm/turbo.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+        "node_modules/**"
```
Comment (janjakubnanista) [inline packages/oapp-evm/turbo.json:12]: This as well, `node_modules` is mos def not an output for the build task

**Review by janjakubnanista — CHANGES_REQUESTED** [review]
Tiny tiny changes

File: packages/oapp-evm/package.json [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  "devDependencies": {
+    "@layerzerolabs/lz-evm-messagelib-v2": "^2.3.29",
+    "@layerzerolabs/lz-evm-protocol-v2": "^2.3.29",
+    "@layerzerolabs/lz-evm-v1-0.7": "^2.3.29",
+    "@layerzerolabs/test-devtools-evm-foundry": "~0.2.8",
+    "@layerzerolabs/toolbox-foundry": "^0.1.7",
+    "@layerzerolabs/toolbox-hardhat": "~0.2.40",
+    "@openzeppelin/contracts": "^5.0.2",
+    "@openzeppelin/contracts-upgradeable": "^5.0.2",
+    "solidity-bytes-utils": "^0.8.0"
```
Comment (ryandgoulding) [inline packages/oapp-evm/package.json:44]: grab this from toolbox-foundry instead.  can use this in foundry.toml:

```
'solidity-bytes-utils/contracts/=node_modules/@layerzerolabs/toolbox-foundry/lib/solidity-bytes-utils/',
```

File: packages/oapp-evm/package.json [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  },
+  "peerDependencies": {
+    "@layerzerolabs/lz-evm-messagelib-v2": "^2.3.29",
+    "@layerzerolabs/lz-evm-protocol-v2": "^2.3.29",
+    "@layerzerolabs/lz-evm-v1-0.7": "^2.3.29",
+    "@openzeppelin/contracts": "^4.8.1 || ^5.0.0",
+    "@openzeppelin/contracts-upgradeable": "^4.8.1 || ^5.0.0",
+    "solidity-bytes-utils": "^0.8.0"
```
Comment (ryandgoulding) [inline packages/oapp-evm/package.json:52]: drops this

**Review by janjakubnanista — APPROVED** [review]

**Comment by Maliksb11** [comment]
ok

**Comment by Maliksb11** [comment]
ok

**Pull request merged**